pkg/ccl/crosscluster/logical/logical_replication_writer_processor.go

```diff
@@ -983,6 +987,9 @@ func (t *txnBatch) HandleBatch(
 	stats := batchStats{}
 	var err error
 	if len(batch) == 1 {
+		if t.spec.Discard == jobspb.LogicalReplicationDetails_DiscardAllDeletes && len(batch[0].KeyValue.Value.RawBytes) == 0 {
```

Comment:
i know discussed this, but just to confirm, we can distinguish between a null pk and a delete because a null pk's have `len(rawbytes) > 0`, right?

Reply:
What do you mean by NULL pk? If you mean a row in which all of the columns stored in the value are NULL, then yes, len(rawBytes) will still be > 0 because we'll encode an empty tuple().

Reply:
Yeah, nil RawBytes is always a delete.